### DISCUSSION TITLE
PYBIND11_MASTER_PROJECT Can the cmake variable be updated?

### DISCUSSION THREAD

**Discussion by messerb5467** [pybind/pybind11#4159] [Q&A] [answered]
@rwgk @henryiii @Skylion007 How hard would it be to change PYBIND11_MASTER_PROJECT to something else like PYBIND11_MAIN_PROJECT or PYBIND11_PRIMARY_PROJECT? This is a usage in conflict with the words matter updates on our side, but we're unsure of what the full usage is in the pybind11 project and would like to gather more information.

**Comment by henryiii** [comment]
Pretty sure that's very much a private variable for us. It's a way for our CMake file to tell if it's the currently running project or if it's running as a submodule. While we have enough users that anything is possible, people should in theory not be reading or writing to that variable.

**Comment by messerb5467** [comment]
Understood. I leave the decision up to you as to whether you want to make the change or not, but would very much appreciate if you would provide an answer either way. There are additional hits with `master project`, `master toctree` and `master_doc` as well that you might want to consider in such a decision.

**Comment by rwgk** [comment, marked as answer]
Based on what @henryiii wrote about github auto-redirects from master to main, I'm in making the move soon, anytime.

I think it'll be the least effort to get the move from master to main done first, then mop up the rest straightaway, or even in the same step.

**Comment by messerb5467** [comment]
This sounds good to me. Thanks so much!
